Mathlib/Probability/Distributions/Exponential.lean

```diff
@@ -0,0 +1,251 @@
+lemma lintegral_split_bounded {y z : ℝ} (f : ℝ → ENNReal) (hzy : z ≤ y) :
+    ∫⁻ x in Iic y, f x = (∫⁻ x in Iio z, f x) + ∫⁻ x in Icc z y, f x := by
+  rw [← Iio_union_Icc_eq_Iic hzy, lintegral_union measurableSet_Icc]
+  rw [Set.disjoint_iff]
+  rintro x ⟨h1 : x < _, h2, _⟩
+  linarith
+
+lemma lintegral_split (f : ℝ → ENNReal) (c : ℝ) :
+    ∫⁻ x, f x = (∫⁻ x in {x | x ≥ c}, f x) + ∫⁻ x in {x | x < c}, f x := by
+  have union : univ = {x: ℝ | x ≥ c} ∪ {x : ℝ | x < c} := by
+    ext x; simp [le_or_lt]
+  have : IsOpen {x : ℝ | x < c} := by exact isOpen_gt' c
+  calc
+  ∫⁻ x, f x = ∫⁻ x in univ, f x ∂ volume := (set_lintegral_univ f).symm
+  _ = ∫⁻ x in {x | x ≥ c} ∪ {x | x < c} , f x ∂ volume := by rw [← union]
+  _ = _ := by
+    apply lintegral_union this.measurableSet
+    rw [Set.disjoint_iff]
+    rintro x ⟨hxge : x ≥ _, hxlt : x < _⟩
+    linarith
+
+namespace ProbabilityTheory
+
+section ExponentialPdf
+
+/-- Define the PDF of the exponential distribution depending on its rate-/
+noncomputable
+def exponentialPdfReal (r x : ℝ) : ℝ :=
+  if 0 ≤ x then r * exp (-(r * x)) else 0
+
+/-- The PDF of the exponential Distribution on the extended real Numbers-/
+noncomputable
+def exponentialPdf (r x : ℝ) : ℝ≥0∞ :=
+  ENNReal.ofReal (exponentialPdfReal r x)
+
+lemma exponentialPdf_eq (r x : ℝ) :
+    exponentialPdf r x = ENNReal.ofReal (if 0 ≤ x then r * exp (-(r * x)) else 0) := rfl
+
+lemma antiDeriv_expDeriv_pos' {r x : ℝ} (hr : 0 < r) :
+    HasDerivAt (fun a ↦ -1/r * exp (-(r * a))) (exp (-(r * x))) x := by
+  convert (((hasDerivAt_id x).const_mul (-r)).exp.const_mul (-1/r)) using 1 <;> field_simp
+
+/-- the Lebesgue-Integral of the exponential PDF over nonpositive Reals equals 0-/
+lemma lintegral_nonpos {x r : ℝ} (hx : x ≤ 0) :
+    ∫⁻ y in Iio x, exponentialPdf r y = 0 := by
+  rw [set_lintegral_congr_fun (g := fun _ ↦ 0) measurableSet_Iio]
+  · rw [lintegral_zero, ← ENNReal.ofReal_zero]
+  · simp only [exponentialPdf_eq, ge_iff_le, ENNReal.ofReal_eq_zero]
+    apply ae_of_all _ fun a (_ : a < _) ↦ by rw [if_neg (by linarith)]
+
+/-- The exponential pdf is measurable. -/
+lemma measurable_exponentialPdfReal (r : ℝ) :
+    Measurable (exponentialPdfReal r) := by
+  refine Measurable.ite ?hp ((measurable_id'.const_mul r).neg.exp.const_mul r) ?hg
+  · exact measurableSet_Ici
+  · exact measurable_const
+
+/-- The exponential Pdf is strongly measurable -/
+lemma stronglyMeasurable_exponentialPdfReal (r : ℝ) :
+    StronglyMeasurable (exponentialPdfReal r) :=
+  (measurable_exponentialPdfReal r).stronglyMeasurable
+
+/-- the exponential Pdf is positive for all positive reals-/
+lemma exponentialPdfReal_pos {x r : ℝ} {hr : 0 < r} (hx : 0 < x) :
+    0 < exponentialPdfReal r x := by
+  simp only [exponentialPdfReal, if_pos hx.le]
+  positivity
+
+/-- The exponential Pdf is nonnegative-/
+lemma exponentialPdfReal_nonneg {r : ℝ} (hr : 0 < r) :
+    ∀ x : ℝ, 0 ≤ exponentialPdfReal r x := by
```

Comment:
Hypotheses/arguments before the `:` instead of in a \forall
```suggestion
lemma exponentialPdfReal_nonneg {r : ℝ} (hr : 0 < r) (x : ℝ) :
    0 ≤ exponentialPdfReal r x := by
```